### DISCUSSION TITLE
Spotify very unstable on BMW ID6

### DISCUSSION THREAD

**Discussion by aidasb123** [BimmerGestalt/AAIdrive#484] [Q&A]
I'm using AAIdrive standard version: 1.3.1 and latest myBMW app. Samsung S10e phone connects with a car with no issues. I can see a Spotify icon in the medis list under FM radio etc. Music continues playing from last time. When I press on Spotify icon 8/10 times it won't open. Can't skip to next song with steering control or radio buttons. Songs keeps playing one after other with no issues.
Sometimes if I can open Spotify and able to skip fee songs, the spofity will stuck with last song logo and time of the song not increase and not possible to skip to the next song. 

Is there any settings in Bluetooth of AAIdrive app I can change get it working?
Also the AAIdrive app showing that my BMW is running ID5 but I know that it's ID6 as it 2018 G30 model.

Regards,

**Comment by hufman** [comment]
Can you check if the controls work better if the AAIdrive app is open on your phone screen? This sounds like the background battery-saving app pausing that I've heard of on Huawei phones. Check your phone's battery saving and background execution options to see if there's something to change there.

**Reply by aidasb123** [reply]
Hi, AAIdrive is added into the "Never sleeping list". I have changed some Bluetooth setting in the Developer mode (Bluetooth AVRCP version: set to 1.6 was 1.4-default) and (Bluetooth PBAP properties: set to 1.1 was 1.2).
It looks like it's running much smoother. But still time to time can't open Spotify app and change tracks.

**Reply by Vasmares** [reply]
@hufman This is the sqme behaviour i described in an earlier issue. It was resolved temporarly but is back again.

Whats curious, is that it is gone for ~ 2weeks after uninstalling and then reinstalling the app.
This extends the issue that the AAIDrive app isnt "connected" to the car.
Bluetooth works fine (phone etc...) but the app states that the app connection is not established.
This also is fixed by reinstalling the app.

Is there a way one would be able to see the exceptions that might be thrown ?
Or are those only in the log files ?

**Reply by hufman** [reply]
It sounds like there are two issues you are talking about @Vasmares : The issue in this thread, about control of a music app being unreliable, and difficulties establishing the car connection through MyBMW.

Relating to this thread's issue, please check if the Notification app is also responding poorly: The same background processing limitation would also be applied to the Notifications app as well as the Music app, and would be a helpful datapoint to know. Do other music apps work better? Sometimes the Spotify connection is flaky, but other music apps use a more reliable standard method of control. Did you confirm whether control works reliably while the phone screen is open and displaying AAIdrive?

The initial connection is trickier to debug. Did you confirm in the car that the Apps connection displays as connected? MyBMW must establish the connection to the car before AAIdrive can add its apps to the car.

Any app crashes are automatically reported to Sentry, if the user has enabled analytics and error reporting. Otherwise exceptions and other log entries are only visible in `adb logcat`.

**Reply by aidasb123** [reply]
App connects to MyBMW app and the car every time. So no issues with connection. Notifications looks like causing even more unstable issues with Spotify. 
As per my reply with a settings on the developer mode, looks like I have no more issues with Spotify. I have enabled notifications also, as previously caused issues. I will keep an eye on this. Think only had once when Spotify wasn't responding to steering wheel/radio buttons to navigate between tracks.
I haven't try different music apps yet.
Thanks guys for help

**Comment by aidasb123** [comment]
![Screenshot_20220101-162039_AAIdrive](https://user-images.githubusercontent.com/96879594/147855015-19265b94-611f-4c21-905f-204d331a87de.jpg)

**Comment by Kay0002** [comment]
Is there a way to view the tracks within a playlists from iDrive?

Currently when I select a playlist or artist from the iDrive app, it auto plays. Was wondering if spotify has the ability to view all of the tracks and choose which to play from iDrive6 (similar to Spotify app in iDrive 7).

Thank you!

**Reply by hufman** [reply]
I believe we opted for the Android Auto behavior of Spotify, where it plays a Browse selection in fewer clicks but then offers a Now Playing view to navigate within the playlist. I'm not familiar with the IDrive7 app's behavior.
Does the Now Playing button show your playlist's contents?

**Reply by Kay0002** [reply]
Hi @hufman 

Ok, yes I do see Now Playing and it shows the list of songs once the playlist is playing. I hadn't noticed this menu option before.

If I may suggest a possible future capability, I think being able to search the titles in the playlist prior to choosing that playlist would be nice (since you have to redo the search process once it begins playing).

Excellent work with your app and thanks again!
